File: MigratoryBirds.java
import java.util.*;

public class MigratoryBirds {
    public static int migratoryBirds(List<Integer> arr) {
        Map<Integer, Integer> birdsMap = new HashMap<>();
        for (Integer birdType : arr) {
            Integer count = birdsMap.get(birdType);
            if (count == null) {
                count = 0;
            }
            birdsMap.put(birdType, count + 1);
        }

        Optional<Map.Entry<Integer, Integer>> maxEntry =
            birdsMap.entrySet().stream()
            .max(Map.Entry.comparingByValue());

        if (maxEntry.isPresent()) {
            return maxEntry.get().getKey();
        } else {
            throw new IllegalStateException("No birds found");
        }
    }

    public static void main(String[] args) {
        List<Integer> arr = Arrays.asList(1, 2, 2, 3, 3, 3);
        int result = migratoryBirds(arr);
        System.out.println(result); // expected output: 3
    }
}

File: JavaScript/MigratoryBirds.js
function migratoryBirds(arr) {
    let birdsMap = {};
    for (let i = 0; i < arr.length; i++) {
        let birdType = arr[i];
        let count = birdsMap[birdType];
        if (count === undefined) {
            count = 0;
        }
        birdsMap[birdType] = count + 1;
    }

    let lowestType = Number.MAX_SAFE_INTEGER;
    let highestCount = 0;
    for (let birdType in birdsMap) {
        if (birdsMap.hasOwnProperty(birdType)) {
            let count = birdsMap[birdType];
            if (count > highestCount || (count === highestCount && birdType < lowestType)) {
                highestCount = count;
                lowestType = birdType;
            }
        }
    }

    return lowestType;
}

console.log(migratoryBirds([1, 2, 5, 8, 8, 2, 6, 5, 2, 9]))
console.log(migratoryBirds([1, 2, 3, 4, 5, 4, 3, 2, 1, 3, 4]))